src/lib/selections.ts

```diff
@@ -694,9 +795,20 @@ export function updateSelections(
       const nodeMeta = getNodeFromPath<Expr>(ast, pathToNode)
       if (err(nodeMeta)) return undefined
       const node = nodeMeta.node
+      const selection = prevSelectionRanges.codeBasedSelections[Number(index)]
```

Comment:
How come `index` isn't a number in the first place?

Reply:
It is a number,
![image](https://github.com/user-attachments/assets/8eeb8947-5b4b-464f-872d-a72f43f32411)

it's just that type is lost from `Object.entries`

Reply:
Does the same thing happen when you `for(const key in nodemap)` ?

Reply:
```
(base) lee@ls76: ~
$ node
Welcome to Node.js v21.4.0.
Type ".help" for more information.
> let a = { 1: 'a', 2: 'b', 3: 'c' }
undefined
> for (const key in a) { console.log(key); }
1
2
3
undefined
```

Reply:
Yeah key is still typed as a string as far as ts is concerned.

Reply:
![image](https://github.com/user-attachments/assets/575b1ce7-1145-4839-a5a9-9aaeb0ec0e81)

Reply:
lame-o